release-notes.txt

```diff
@@ -1,3 +1,550 @@
+    * drivers/ds18: fix `ds18_read` for negative temperatures
+    * drivers/sx127x: uncrustify main file
+    + sx127x: add several NETOPT for GNRC LoRaWAN
+    * drivers/sx127x: fix device reset
+    + drivers: stm32 eth peripheral driver
+    - sx127x: remove MULTIDIO and fix missing interrupt handling
+    + drivers/periph/gpio_util: add gpio_util_shiftin()
+
+Build System / Tooling
+----------------------
+    * makefiles/docker: export BOARDS
+    + makefiles: add bootloaders to the list of applications
+    * Makefile.include: allow overwriting flash-recipe
+    * dist/tools/buildsystem_sanity_check: add an export variable check
+    * sys/Makefile.include: include riotboot headers when
+      FEATURES_REQUIRED=riotboot
+    + make: add print-versions helper target
+    + make: add LOG_LEVEL to overridable variables
+    * riotboot/Makefile.include: increase RIOTBOOT_HDR_LEN for ARMv7*-M
+    * dist/tools/build_system_sanity_check: BUG fix errors being ignored
+    * boards/common: use Makefile.features
+    + make: add termdeps target
+    * periph_common: add as dependency to periph drivers
+    + nrf52: Add USB peripheral driver
+    + makefiles/boards.inc.mk: list boards variables
+    + makefiles/app_dirs.inc.mk: target to list supported applications/boards
+    + Makefile.features: add a common file for the features parsing
+    + Makefile.features: add declarative FEATURES_ variables definition
+    + makefiles/utils/variables: add functions to help managing variables
+    * Migrate all boards to define FLASHFILE
+    * make: remove exports for flash debug reset
+    * board: remove exporting PORT_LINUX|PORT_DARWIN
+    * tools/openocd.sh: try to probe the board for real flash address
+
+Testing
+-------
+    * tests/periph_flashpage: add RWWEE automatic test if hw supports it
+    * tests/pthread_tls: allow negative key values
+    * tests/gnrc_udp: include `gnrc_pktbuf_cmd` per default
+    + tests/periph_hwrng: add automated python test
+    * tests/periph_timer: include kw41z boards in low-power timer boards
+    * tests/pkg_c25519: increase timeout for arduino-mega2560
+    + tests/xtimer_usleep: fail with negative offsets
+    * tests/unittests: remove unused LARGE_STACK_TESTS
+    + unittests: add tsrb tests
+    + tests/stdin: add non regression test for stdin module
+    + tests/xtimer_mutex_lock_timeout: add simple case test
+    + Add NimBLE heart rate sensor example (GATT notifications)
+    * tests/driver_rn2xx3: fix invalid element index for txmode
+    * tests/pkg_semtech-loramac: don't init the mac from main
+    * tests/devfs: move tests-devfs out of unittests
+    + examples/lorawan: fix semtech_loramac_send TX ret code
+    * tests/unittests: split remaining packages tests to different test
+                       directory
+
+API Changes
+========================
+- uart_stdio RX is now optional
+- CPU_MODEL declared in boards/Makefile.features
+
+
+```c
```

Comment:
Do we have to be as detailed? and isn't this rather

~~~md
```diff
~~~

?

Reply:
Well I wanted the c syntax highlighted but I can also go with diff if you prefer.  I think I would like to note this though.

Reply:
kk